code/modules/mob/living/silicon/robot/robot_defense.dm

```diff
@@ -79,75 +79,76 @@
-				emag_cooldown = world.time
-				return
-
-			var/ai_is_antag = 0
-			if(connected_ai && connected_ai.mind)
-				if(connected_ai.mind.special_role)
-					ai_is_antag = (connected_ai.mind.special_role == "traitor")
-			if(ai_is_antag)
-				user << "<span class='notice'>You emag [src]'s interface.</span>"
-				src << "<span class='danger'>ALERT: Foreign software execution prevented.</span>"
-				connected_ai << "<span class='danger'>ALERT: Cyborg unit \[[src]] successfuly defended against subversion.</span>"
-				log_game("[key_name(user)] attempted to emag cyborg [key_name(src)] slaved to traitor AI [connected_ai].")
-				emag_cooldown = world.time
-				return
-
-			if(wiresexposed)
-				user << "<span class='warning'>You must unexpose the wires first!</span>"
-				return
-			else
-				emag_cooldown = world.time
-				sleep(6)
-				SetEmagged(1)
-				SetLockdown(1) //Borgs were getting into trouble because they would attack the emagger before the new laws were shown
-				lawupdate = 0
-				connected_ai = null
-				user << "<span class='notice'>You emag [src]'s interface.</span>"
-				message_admins("[key_name_admin(user)] emagged cyborg [key_name_admin(src)].  Laws overridden.")
-				log_game("[key_name(user)] emagged cyborg [key_name(src)].  Laws overridden.")
-				clear_supplied_laws()
-				clear_inherent_laws()
-				clear_zeroth_law(0)
-				laws = new /datum/ai_laws/syndicate_override
-				var/time = time2text(world.realtime,"hh:mm:ss")
-				lawchanges.Add("[time] <B>:</B> [user.name]([user.key]) emagged [name]([key])")
-				set_zeroth_law("Only [user.real_name] and people they designate as being such are Syndicate Agents.")
-				src << "<span class='danger'>ALERT: Foreign software detected.</span>"
-				sleep(5)
-				src << "<span class='danger'>Initiating diagnostics...</span>"
-				sleep(20)
-				src << "<span class='danger'>SynBorg v1.7 loaded.</span>"
-				sleep(5)
-				src << "<span class='danger'>LAW SYNCHRONISATION ERROR</span>"
-				sleep(5)
-				src << "<span class='danger'>Would you like to send a report to NanoTraSoft? Y/N</span>"
-				sleep(10)
-				src << "<span class='danger'>> N</span>"
-				sleep(20)
-				src << "<span class='danger'>ERRORERRORERROR</span>"
-				src << "<b>Obey these laws:</b>"
-				laws.show_laws(src)
-				src << "<span class='danger'>ALERT: [user.real_name] is your new master. Obey your new laws and their commands.</span>"
-				SetLockdown(0)
-				update_icons()
+	if(user == src)//To prevent syndieborgs from emagging themselves
+		return
+	if(!opened)//Cover is closed
+		if(locked)
+			user << "<span class='notice'>You emag the cover lock.</span>"
+			locked = 0
+		else
+			user << "<span class='warning'>The cover is already unlocked!</span>"
+		return
+	if((world.time - 100) < emag_cooldown)
+		return
+	if(wiresexposed)
+		user << "<span class='warning'>You must unexpose the wires first!</span>"
+		return
+
+	user << "<span class='notice'>You emag [src]'s interface.</span>"
+	emag_cooldown = world.time
```

Comment:
can you change the check above to be world.time < emag_cooldown

and then change this line to emag_cooldown = world.time + 100

having them backwards is very confusing